rules/group.go

```diff
@@ -643,28 +650,32 @@ func (g *Group) Eval(ctx context.Context, ts time.Time) {
 	if ctrl == nil {
 		ctrl = sequentialRuleEvalController{}
 	}
-	for _, batch := range ctrl.SplitGroupIntoBatches(ctx, g) {
-		for _, ruleIndex := range batch {
-			select {
-			case <-g.done:
-				return
-			default:
-			}
-
-			rule := g.rules[ruleIndex]
-			if len(batch) > 1 && ctrl.Allow(ctx, g, rule) {
-				wg.Add(1)
 
-				go eval(ruleIndex, rule, func() {
-					wg.Done()
-					ctrl.Done(ctx)
-				})
-			} else {
-				eval(ruleIndex, rule, nil)
+	batches := ctrl.SplitGroupIntoBatches(ctx, g)
+	if batches == nil || len(batches) == len(g.rules) {
```

Comment:
I think we should scope out this look into a separate function so that we can unit test it - I'd love to feed it different outcomes of `SplitGroupIntoBatches` (e.g. no batch, batch != g.rules, both 0 batch and 0 rules, 1 batch and 1 rule, 3 batch and 3 rules, 3 batch and 5 rules and 5 batch and 3 rules) just to ensure this does what we think it does in the correct way.

Reply:
I think the condition was just unnecessary and error prone. Didn't consider batches of 0 rules as well.
I've replaced it with just a zero check

So the possible options are:
- Empty or nil batches slice: Fully sequential (as before)
- Encounter a batch with more than 1 rule: goroutines
- Encounter a batch with only 1 rule: executed sequentially